src/main/java/duke/local/DataManager.java

```diff
@@ -0,0 +1,124 @@
+package duke.local;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Scanner;
+
+import duke.task.TaskManager;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+public class DataManager {
+
+    private final String filePath;
+    private final File dataFile;
+
+    private final String FILE_SEPARATOR_REGEX = "\\s\\|\\s";
+    private final String TODO_FILE_FORMAT_REGEX = "T\\s\\|\\s[01]\\s\\|.+";
+    private final String DEADLINE_FILE_FORMAT_REGEX = "D\\s\\|\\s[01]\\s\\|.+\\s\\|.+";
+    private final String EVENT_FILE_FORMAT_REGEX = "E\\s\\|\\s[01]\\s\\|.+\\s\\|.+";
+
+    private final String TODO_TYPE = "T";
+    private final String DEADLINE_TYPE = "D";
+    private final String EVENT_TYPE = "E";
+
+    private final String FILE_NOT_FOUND_MSG = "File does not exist!";
+    private final String FILE_CREATION_ERROR_MSG = "File cannot be created!";
+    private final String FILE_WRITE_ERROR_MSG = "File cannot be written to!";
+
+    public DataManager(String filePath) {
+        this.filePath = filePath;
+        this.dataFile = new File(filePath);
+    }
+
+    public void printFileContents() throws FileNotFoundException {
+        Scanner in = new Scanner(dataFile);
+
+        while (in.hasNext()) {
+            System.out.println(in.nextLine());
+        }
+    }
+
+    public void createFile() throws IOException {
+        Files.createDirectories(Path.of(dataFile.getParent()));
+        dataFile.createNewFile();
+    }
+
+    public TaskManager loadDataFromFile() {
+
+        TaskManager taskMgr = new TaskManager();
+
+        try {
+
+            Scanner in = new Scanner(dataFile);
+
+            while (in.hasNext()) {
+
+                String task = in.nextLine();
+                String[] taskArgs = task.split(FILE_SEPARATOR_REGEX, 4);
+
+                String taskType = taskArgs[0];
+                boolean isDone = false;
+
+                if (taskArgs[1].equals("1")) {
+                    isDone = true;
+                }
+
+                String taskDescription = taskArgs[2];
+                String datetime = "";
```

Comment:
The variable name `datetime` is a bit ambiguous (not really clear from the name what it means) and also since its two separate words follow camelCase